netweaver/monitoring.sls

```diff
@@ -6,33 +6,30 @@ prometheus_sap_host_exporter_pkg:
     - name: prometheus-sap_host_exporter
 
 # the sid, instance number pair of a node is unique, so we need to adapt configuration
-{% for node in netweaver.nodes if host == node.host and node.sap_instance != "db" %}
-{% set instance = '{:0>2}'.format(node.instance) %}
-{% set daemon_instance = '{}_{}'.format(node.sid, instance) %}
+{% for node in netweaver.nodes if node.sap_instance != "db" %}
+{% set sap_instance_nr = '{:0>2}'.format(node.instance) %}
+{% set exporter_instance = '{}_{}{}'.format(node.sid, node.sap_instance.upper(), sap_instance_nr) %}
 
-sap_host_exporter_service_{{ daemon_instance }}:
-  service.running:
-    - name: prometheus-sap_host_exporter@{{ daemon_instance }}
-    - enable: True
-    - restart: True
+# we bind each exporter instance to a SAP instance virtual host
+sap_host_exporter_configuration_{{ exporter_instance }}:
+  file.managed:
+    - name: /etc/sap_host_exporter/{{ exporter_instance }}.yaml
+    - contents: |
+         address: {{ node.virtual_host }}
+         sap-control-uds: /tmp/.sapstream5{{ sap_instance_nr }}13
```

Comment:
Is this `/tmp` file usage reliable? I mean, the content in this folder is removed periodically. Can we trust that this solution will work long term?

Reply:
This is how SAP documents UDS usage. I think the socket files are created by the SAP Start Service.